device/mockup/tt_mockup_device.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "device/tt_cluster_descriptor.h"
+#include "device/tt_device.h"
+
+class tt_MockupDevice : public tt_device {
+   public:
+    tt_MockupDevice(const std::string& sdesc_path) : tt_device(sdesc_path) {
+        soc_descriptor_per_chip.emplace(0, tt_SocDescriptor(sdesc_path));
+        std::set<chip_id_t> target_devices = {0};
+    }
+    virtual ~tt_MockupDevice() {}
+
+    // Setup/Teardown Functions
+    virtual std::unordered_map<chip_id_t, tt_SocDescriptor>& get_virtual_soc_descriptors() override {
+        return soc_descriptor_per_chip;
+    }
+    virtual void set_device_l1_address_params(const tt_device_l1_address_params& l1_address_params_) override{};
+    virtual void set_device_dram_address_params(const tt_device_dram_address_params& dram_address_params_) override{};
+    virtual void set_driver_host_address_params(const tt_driver_host_address_params& host_address_params_) override{};
+    virtual void set_driver_eth_interface_params(
+        const tt_driver_eth_interface_params& eth_interface_params_) override{};
+    virtual void start_device(const tt_device_params& device_params) override {}
+    virtual void assert_risc_reset() override{};
```

Comment:
nit: change `override{};` to `override {}`, like for `start_device` function. Diff is semicolon (it is not needed), and space between `override` and `{}`

Reply:
I hope that automatic linter will solve this in the future

Reply:
thanks - will do. 

btw. this was with automatic linter from tt-metal 😅